### PR TITLE
Holzapfel Ogden Modified Anisotropy (HO-ma) Constitutive Model

### PR DESCRIPTION
**Pull request by divyaadil23**
Same as the previous pull request, but the commits have been squashed into one.

## Current situation
https://github.com/SimVascular/svFSIplus/issues/249

## Release Notes 

- Added the Holzapfel Ogden Modified Anisotropy (HO-ma) Constitutive Model to material models.
- Updated Holzapfel Ogden Model.
- Both HO and HO-ma models are taken from https://github.com/vvedula22/svFSI/blob/master/Code/Source/svFSI/MATMODELS.f and modified. Added additional terms containing the second derivative of smoothed heaviside funciton to the elasticity tensor to make it exact. Changed the approximate derivatives of heaviside function to exact. Removed fds (dot product of fiber directions in 8th invariant).
- Tested it with the new framework of unit tests - the unit tests now test the order of convergence for finite difference estimates.
- Added some integrated tests for struct and ustruct for both models.

## Documentation
 
Where can I document how to use the new material model? Also, should I add the derivation?

## Testing
 
Added 4 integration tests for HO and HO-ma models. Added new unit tests for both.

## Code of Conduct & Contributing Guidelines 
- [x] I agree to follow the [Code of Conduct](https://github.com/SimVascular/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/SimVascular/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aabrown100-git** [comment]
Looks like in your most recent commit (merging SimVascular::main), your new tests failed. I think you might just have to rename your new test input files from svFSI.xml to svFSIplus.xml.

**Review by aabrown100-git — CHANGES_REQUESTED** [review]

File: Code/Source/svFSI/ComMod.h [hunk 1]
```diff
@@ -333,6 +333,9 @@ class fibStrsType
     // Constant steady value
     double g = 0.0;
 
+    // Cross fiber stress parameter
+    double eta_s = 0.0;
```
Comment (aabrown100-git) [inline Code/Source/svFSI/ComMod.h:337]: Is the user able to set this parameter in the input file?

**Review by aabrown100-git — CHANGES_REQUESTED** [review]

File: Code/Source/svFSI/ComMod.h (hunk 1 above)
Comment (divyaadil23) [reply Code/Source/svFSI/ComMod.h:337]: No, they can't do it in the input file as of now.

File: tests/cases/fluid/pipe_RCR_3d_trilinos_bj/README.md [hunk 2]
```diff
@@ -6,6 +6,17 @@ Simulate unsteady fluid flow in a pipe.
 The simulation differs from the <a href="https://github.com/SimVascular/svFSIplus/tree/main/tests/cases/fluid/pipe_RCR_3d"> Fluid RCR 3D Pipe </a> test only in the linear algebra package it uses.
 
 The simulation uses the Trilinos linear algebra package with a **trilinos-blockjacobi** preconditioner.
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (divyaadil23) [reply tests/cases/fluid/pipe_RCR_3d_trilinos_bj/README.md:9]: I think I didn't accept the incoming change for this file from Simvascular::main. Doing it now.

File: Code/Source/svFSI/ComMod.h (hunk 1 above)
Comment (aabrown100-git) [reply Code/Source/svFSI/ComMod.h:337]: Okay, we can leave that for a future pull request.

**Review by msbazzi — APPROVED** [review]

File: Code/Source/svFSI/mat_fun_carray.h [hunk 3]
```diff
@@ -458,6 +458,15 @@ double norm(const Vector<double>& u, const double v[N])
   return norm;
 }
 
+template <size_t N>
+double norm(const Vector<double>& u, const Vector<double>& v)
```
Comment (ktbolt) [inline Code/Source/svFSI/mat_fun_carray.h:462]: There is already a `norm` function in `utils.cpp`.

**Review by zinanhu0810 — APPROVED** [review]

**Comment by aabrown100-git** [comment]
Looks like your HO and HO-ma integration tests failed. It's probably because the reference solutions were generated with the default `solid_viscosity` value, which was erroneously set to 0.9 previously. The most recent [commit](https://github.com/SimVascular/svFSIplus/commit/aca9a10dc5dce2614355ce5b8a755c2eaaa13a37) changes the default value to 0.0, so the integration tests now use a `solid_viscosity` value of 0.0.

You can verify this by setting the `solid_viscosity` parameter to 0.9 in your test input files
`
<Solid_viscosity> 0.9 </Solid_viscosity>
`
The tests should pass. If so, I think you can just rerun your tests with 0 solid viscosity (just delete it from the input file) and update your reference solution with the result.

File: Code/Source/svFSI/mat_fun_carray.h (hunk 3 above)
Comment (divyaadil23) [reply Code/Source/svFSI/mat_fun_carray.h:462]: I am removing this function.

File: tests/cases/struct/LV_HolzapfelOgdenModifiedAnisotropy_passive/svFSIplus.xml [hunk 4]
```diff
@@ -58,9 +58,8 @@
    <Density> 1.0 </Density>                           <!-- g/cm^3 -->
    <Elasticity_modulus> 1.0e6 </Elasticity_modulus>   <!-- dyne/cm^2 -->
    <Poisson_ratio> 0.483333 </Poisson_ratio>
-   <Viscosity model="Constant" >
-         <Value> 0.0 </Value>
-      </Viscosity>
+   
+   <!-- <Solid_viscosity> 0.9 </Solid_viscosity> -->
```
Comment (aabrown100-git) [inline tests/cases/struct/LV_HolzapfelOgdenModifiedAnisotropy_passive/svFSIplus.xml:62]: Small thing, could you remove this commented Solid_viscosity line? I just submitted a pull request updating solid viscosity and this syntax will no longer be valid.

File: tests/cases/struct/LV_HolzapfelOgden_passive/svFSIplus.xml [hunk 5]
```diff
@@ -58,9 +58,8 @@
    <Density> 1.0 </Density>                           <!-- g/cm^3 -->
    <Elasticity_modulus> 1.0e6 </Elasticity_modulus>   <!-- dyne/cm^2 -->
    <Poisson_ratio> 0.483333 </Poisson_ratio>
-   <Viscosity model="Constant" >
-         <Value> 0.0 </Value>
-      </Viscosity>
+   
+   <!-- <Solid_viscosity> 0.9 </Solid_viscosity> -->
```
Comment (aabrown100-git) [inline tests/cases/struct/LV_HolzapfelOgden_passive/svFSIplus.xml:62]: Remove this too

File: tests/cases/ustruct/LV_HolzapfelOgden_passive/svFSIplus.xml [hunk 6]
```diff
@@ -56,9 +56,8 @@
    <Density> 1.0 </Density>                           <!-- g/cm^3 -->
    <Elasticity_modulus> 1.0e6 </Elasticity_modulus>   <!-- dyne/cm^2 -->
    <Poisson_ratio> 0.483333 </Poisson_ratio>
-   <Viscosity model="Constant" >
-         <Value> 0.0 </Value>
-      </Viscosity>
+   
+   <!-- <Solid_viscosity> 0.9 </Solid_viscosity> -->
```
Comment (aabrown100-git) [inline tests/cases/ustruct/LV_HolzapfelOgden_passive/svFSIplus.xml:60]: Remove this too

File: README.md
Comment (aabrown100-git) [inline README.md:1]: These changes seem like they're reintroducing typos that had been fixed. I think we should revert changes to this file.

File: tests/cases/fluid/pipe_RCR_3d_bj_trilinos/README.md
Comment (aabrown100-git) [inline tests/cases/fluid/pipe_RCR_3d_bj_trilinos/README.md:1]: This file should be removed

File: tests/cases/fluid/pipe_RCR_3d_ilut_trilinos/README.md
Comment (aabrown100-git) [inline tests/cases/fluid/pipe_RCR_3d_ilut_trilinos/README.md:1]: This file should be removed

File: tests/cases/fsi/pipe_3d_ml_trilinos/README.md
Comment (aabrown100-git) [inline tests/cases/fsi/pipe_3d_ml_trilinos/README.md:1]: This file should be removed

**Pull request merged**